src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -353,39 +360,56 @@ public static FormulaValue Text(FormattingInfo formatInfo, IRContext irContext,
+            if (formatString != null && !TextFormatUtils.IsValidFormatArg(formatString, out textFormatArgs))
             {
                 var customErrorMessage = StringResources.Get(TexlStrings.ErrIncorrectFormat_Func, culture.Name);
                 return CommonErrors.GenericInvalidArgument(irContext, string.Format(CultureInfo.InvariantCulture, customErrorMessage, "Text"));
             }
 
-            var isText = TryText(formatInfo, irContext, args[0], formatString, out StringValue result);
+            var isText = TryText(formatInfo, irContext, args[0], textFormatArgs, out StringValue result);
 
             return isText ? result : CommonErrors.GenericInvalidArgument(irContext, StringResources.Get(TexlStrings.ErrTextInvalidFormat, culture.Name));
         }
 
-        public static bool TryText(FormattingInfo formatInfo, IRContext irContext, FormulaValue value, string formatString, out StringValue result)
+        public static bool TryText(FormattingInfo formatInfo, IRContext irContext, FormulaValue value, TextFormatArgs textFormatArgs, out StringValue result)
         {
             var timeZoneInfo = formatInfo.TimeZoneInfo;
             var culture = formatInfo.CultureInfo;
-            var hasDateTimeFmt = false;
-            var hasNumberFmt = false;
+            CultureInfo formatCulture = culture;
+            var formatString = textFormatArgs.FormatArg;
             result = null;
 
-            if (formatString != null && !TextFormatUtils.IsValidFormatArg(formatString, out hasDateTimeFmt, out hasNumberFmt))
+            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
+
+            if (!string.IsNullOrEmpty(textFormatArgs.FormatCultureName) && !TryGetCulture(textFormatArgs.FormatCultureName, out formatCulture))
             {
                 return false;
             }
 
-            Contract.Assert(StringValue.AllowedListConvertToString.Contains(value.Type));
+            if (!string.IsNullOrEmpty(formatString) && textFormatArgs.HasNumericFmt)
+            {
+                // Get en-US numeric format string.
+                // \uFEFF is the zero width no-break space codepoint. This will be used to swap with number group seperator character.
+                string numberGroupSeperator = "\uFEFF";
+                var numberCultureFormat = formatCulture.NumberFormat;
+
+                formatString = formatString.Replace(numberCultureFormat.NumberGroupSeparator, numberGroupSeperator);
+                if (string.IsNullOrWhiteSpace(numberCultureFormat.NumberGroupSeparator))
+                {
+                    formatString = formatString.Replace(" ", numberGroupSeperator).Replace("\u202F", numberGroupSeperator);
+                }
+
+                formatString = formatString.Replace(numberCultureFormat.NumberDecimalSeparator, ".");
+                formatString = formatString.Replace(numberGroupSeperator, ",");
+            }

```

Comment:
can this be moved into textFormatArgs?
It's ultimately parsing...

Reply:
moved it.